### ISSUE TITLE
K8sContainerContext is missing tolerations, nodeSelector, and podSecurityContext

### ISSUE THREAD

**Issue by craustin** [dagster-io/dagster#9924]
### Dagster version

1.0.11

### What's the issue?

When a run job/pod is created, it doesn't include _all_ the config of the user repo deployment, even if `includeConfigInLaunchedRuns.enabled` is `true`. It is missing `tolerations`, `nodeSelector`, and `podSecurityContext`.

### What did you expect to happen?

I expected all the config specified in the elements of the `deployments` list to be copied to the run jobs/pods, if `includeconfigInLaunchedRuns.enabled` is `true.

We'd need to add those missing properties here:
https://github.com/dagster-io/dagster/blob/f5ea4b7d861d4966166b54bc575b7a3ebbfe7851/python_modules/libraries/dagster-k8s/dagster_k8s/container_context.py#L24

and here:
https://github.com/dagster-io/dagster/blob/4754109a27fd0aedb107e4101818d8e4c174d492/helm/dagster/charts/dagster-user-deployments/templates/helpers/_helpers.tpl#L97

### Deployment type

Dagster Helm chart

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by sryza** [comment]
@johannkm - know what's going on here?

**Comment by johannkm** [comment]
Hi @craustin, yeah those should be added. Feel free to pr, otherwise we can take it

**Comment by craustin** [comment]
Thanks @johannkm. I just put up https://github.com/dagster-io/dagster/pull/9940 to address this. Let me know what you think.

**Comment by paralyzah** [comment]
Probably, I think, isn't enough.
If you decided to dig into it - I think you need add a container security contexts accordingly to docs:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#securitycontext-v1-core

**Comment by craustin** [comment]
Yeah, true. There are more options we could add (like the container security context.) The PR only adds tolerations, nodeSelector, and podSecurityContext, but ideally this would be generalized to make it easy to support more passthrough of k8s config. It's pretty clunky right now.

**Comment by alishchytovych** [comment]
Hi, I've tried to use nodeSelector and tolerations with the latest version 1.4.16 and it doesn't seem to work. Any plans to implement?

**Comment by sverbruggen** [comment]
I am also affected by this issue in 2024. Please have a look at this one :)

**Comment by suyash-811** [comment]
Hey,
we also faced the same issue this week and it would be great if we can pass on the configuration options to the pod created by the run. Is someone working on this currently? If not, i would like to look into this.

**Comment by gita-unlearn** [comment]
I have PR for adding the tolerations to the run 
https://github.com/dagster-io/dagster/pull/25940

**Comment by cmpadden** [comment]
Implemented in https://github.com/dagster-io/dagster/commit/70c198b0004fc51b7b190a0c1fe6ee4a39abffd8
